packages/contracts-bedrock/src/L2/SuperchainTokenBridge.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+// Libraries
+import { Predeploys } from "src/libraries/Predeploys.sol";
+
+// Interfaces
+import { ISuperchainERC20 } from "src/L2/interfaces/ISuperchainERC20.sol";
+import { IL2ToL2CrossDomainMessenger } from "src/L2/interfaces/IL2ToL2CrossDomainMessenger.sol";
+
+/// @custom:proxied true
+/// @custom:predeploy 0x4200000000000000000000000000000000000028
+/// @title SuperchainTokenBridge
+/// @notice The SuperchainTokenBridge allows for the bridging of ERC20 tokens to make them fungible across the
+///         Superchain. It builds on top of the L2ToL2CrossDomainMessenger for both replay protection and domain
+///         binding.
+contract SuperchainTokenBridge {
+    /// @notice Thrown when attempting to perform an operation and the account is the zero address.
+    error ZeroAddress();
+
+    /// @notice Thrown when attempting to relay a message and the function caller (msg.sender) is not
+    /// L2ToL2CrossDomainMessenger.
+    error CallerNotL2ToL2CrossDomainMessenger();
```

Comment:
This looks analagous to the `Unauthorized` eror we use elsewhere in the codebase. Consider using that to minimize the amount of new errors for a simpler ABI. Either the `Unauthorized()` or the `Unauthorized(expected, actual)` signatures that we use are ok with me

Reply:
We discussed this with @skeletor-spaceman, @0xng, and @gotzenx, and we believe that having explicit error messages offers more advantages than disadvantages. However, we're open to changing them if it's absolutely necessary for the PR.

Reply:
I don't feel strongly, I just want to be careful about exploding the amount of custom errors vs. grouping related ones  and using params to get specific. For example you could even do `Unauthorized(string)` to revert with `Unauthorized("Caller is not L2ToL2CrossDomainMessenger")`

Reply:
Returning a string defeats the purpose of using a custom error no?

Reply:
I definitely am a minimalist and prefer a minimal API but if you are very opinionated about very specific custom errors, its ok to ship

Reply:
> Returning a string defeats the purpose of using a custom error no? At least with regards to the bytecode size

Ultimately depends on what we want to optimize for. The tradeoff is roughly:
- Bespoke errors for each case: Biggest ABI, but full rationale is part of the ABI for consumers
- `Unauthorized(address actual, address expected)` types of errors: Small ABI, but ABI alone doesn't tell consumer what each address semantically represents from the ABI (i.e. `CallerNotL2ToL2CrossDomainMessenger` tells me I expected the L2ToL2CrossDomainMessenger, but I have to know that the `expected: 0x...` address is the L2ToL2CrossDomainMessenger address
- `Unauthorized(string)`: Same ABI size, more flexibility in communicating to users what went wrong, larger bytecode and gas due to string usage
- `Unauthorized(bytes32)`: Like the prior bullet, but limited string size, so smaller/cheaper

Personally I also like smaller interfaces (for functions, errors, and events) as it means smaller surface areas (less attack surface when talking about functions), and for me that makes it easier to fit full context in my head when reasoning + less room for mistakes when choosing which to use.

Ultimately whatever we decide should be upstreamed to the Style Guide to avoid this bikeshed in the future 😅

Reply:
We have broken returndata in the past and usually do not consider it a "breaking change" for the ABI given the `require` -> custom error migration. We need to continue migrating to custom errors given our style guide. I would prefer to not bikeshed on this too long, am ok with this going in as is and change it in the future if necessary to standardize. cc @smartcontracts to get on your radar around standardizing style with custom errors

Reply:
Created https://github.com/ethereum-optimism/optimism/issues/12478 to track. Great to know that we are ok with breaking the revert path ABIs, that makes it much less urgent to standardize here